### PR TITLE
Allow to override cluster endpoint

### PR DESCRIPTION
**Pull request by pburgisser**
When there is no bridge (on mac), the wait_for_cluster cannot reach the API as it can't contact the pod IP.
In conjunction with port publishing, terraform can then access the master.

### CODE REVIEW

**Comment by pburgisser** [comment]
cc @mcanevet

**Comment by mcanevet** [comment]
Hmmm... I'm not in favor of adding a `base_domain` variable.
Maybe we should find a way to either ignore wait for cluster or use localhost when the port is published.

**Comment by pburgisser** [comment]
We must be able to change replace the pod IP by something in the kubectl output that is also used when deploying ArgoCD.
Maybe when the network_name isn't "bridge" we replace the container IP by localhost? or we can rename the variable by something like "k8s_master_hostname" ?

**Comment by mcanevet** [comment]
Maybe a `cluster_endpoint` variable that defaults to `https://<container_ip>:6443` but could be overridden.

**Comment by mcanevet** [comment]
Or maybe a `tls_san` variable that we would also pass as argument to the server : https://rancher.com/docs/k3s/latest/en/installation/install-options/server-config/#listeners

**Comment by pburgisser** [comment]
cluster_endpoint sounds good. When set we add to the SAN.
What do you think?

**Comment by mcanevet** [comment]
Yes that sounds good. The tricky part would be to extract the hostname from the endpoint, but it feasible.

**Comment by pburgisser** [comment]
@mcanevet done.
I'm not sure the tls_san is necessary as it works this way already.
I named the variable api_endpoint as cluster_endpoint is already an output.

**Comment by mcanevet** [comment]
@pburgisser i'm pretty sure that if you don't set `tls-san`, the generated certificate will not contain the URL you pass as API endpoint.
Regarding the variable name, I think we should keep  ̀cluster_endpoint` also for input as it's the same thing. The only thing you have to do is to return the `cluster_endpoint` you passed as input if you override it.

**Comment by pburgisser** [comment]
@pburgisser There you go :)
` DNS:192-168-1-118.nip.io, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc.cluster.local, DNS:localhost, IP Address:10.43.0.1, IP Address:127.0.0.1, IP Address:172.17.0.8`

File: outputs.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 output "cluster_endpoint" {
-  value = format("https://%s:6443", docker_container.k3s_server.ip_address)
+  value = format("https://%s:6443", coalesce(var.cluster_endpoint, docker_container.k3s_server.ip_address))
```
Comment (mcanevet) [inline outputs.tf]: `local.cluster_endpoint`?

**Review by mcanevet — CHANGES_REQUESTED** [review]

File: outputs.tf (hunk 1 above)
Comment (pburgisser) [reply outputs.tf]: Damn it, I missed this guy

**Pull request merged**